codec-mqtt/src/main/java/io/netty/handler/codec/mqtt/MqttReasonCodes.java

```diff
@@ -0,0 +1,422 @@
+            this.byteValue = byteValue;
+        }
+
+        /**
+         * @return the value number corresponding to the constant.
+         * */
+        public byte byteValue() {
+            return byteValue;
+        }
+
+        /**
+         * @param b the number to decode.
+         * @return the enum value corresponding to the number.
+         * */
+        public static PubComp valueOf(byte b) {
+            return valueOfHelper(b, VALUES);
+        }
+    }
+
+    /**
+     * Reason codes for MQTT SubAck message.
+     */
+    public enum SubAck {
+        GRANTED_QOS_0((byte) 0x00),
+        GRANTED_QOS_1((byte) 0x01),
+        GRANTED_QOS_2((byte) 0x02),
+        UNSPECIFIED_ERROR((byte) 0x80),
+        IMPLEMENTATION_SPECIFIC_ERROR((byte) 0x83),
+        NOT_AUTHORIZED((byte) 0x87),
+        TOPIC_FILTER_INVALID((byte) 0x8F),
+        PACKET_IDENTIFIER_IN_USE((byte) 0x91),
+        QUOTA_EXCEEDED((byte) 0x97),
+        SHARED_SUBSCRIPTIONS_NOT_SUPPORTED((byte) 0x9E),
+        SUBSCRIPTION_IDENTIFIERS_NOT_SUPPORTED((byte) 0xA1),
+        WILDCARD_SUBSCRIPTIONS_NOT_SUPPORTED((byte) 0xA2);
+
+        private static final SubAck[] VALUES;
+
+        static {
+            SubAck[] values = values();
+            VALUES = new SubAck[163];
+            for (SubAck code : values) {
+                final int unsignedByte = code.byteValue & 0xFF;
+                // Suppress a warning about out of bounds access since the enum contains only correct values
+                VALUES[unsignedByte] = code;    //  [java/index-out-of-bounds]
+            }
+        }
+
+        private final byte byteValue;
+
+        SubAck(byte byteValue) {
+            this.byteValue = byteValue;
+        }
+
+        /**
+         * @return the value number corresponding to the constant.
+         * */
+        public byte byteValue() {
+            return byteValue;
+        }
+
+        /**
+         * @param b the number to decode.
+         * @return the enum value corresponding to the number.
+         * */
+        public static SubAck valueOf(byte b) {
+            return valueOfHelper(b, VALUES);
+        }
+    }
+
+    /**
+     * Reason codes for MQTT UnsubAck message.
+     */
+    public enum UnsubAck {
+        SUCCESS((byte) 0x00),
+        NO_SUBSCRIPTION_EXISTED((byte) 0x11),
+        UNSPECIFIED_ERROR((byte) 0x80),
+        IMPLEMENTATION_SPECIFIC_ERROR((byte) 0x83),
+        NOT_AUTHORIZED((byte) 0x87),
+        TOPIC_FILTER_INVALID((byte) 0x8F),
+        PACKET_IDENTIFIER_IN_USE((byte) 0x91);
+
+        private static final UnsubAck[] VALUES;
+
+        static {
+            UnsubAck[] values = values();
+            VALUES = new UnsubAck[146];
+            for (UnsubAck code : values) {
+                final int unsignedByte = code.byteValue & 0xFF;
+                // Suppress a warning about out of bounds access since the enum contains only correct values
+                VALUES[unsignedByte] = code;    //  [java/index-out-of-bounds]
```

Comment:
Do we still get warnings in these methods, or can we remove the suppression?

Reply:
Honestly I don't know if this is a sort of standard to communicate to all checking tools to avoid this warning notification. In that case worthwhile to keep it, because the added tests does a dynamic check while the tools does a static check.
Let me know if it needs to be removed.